src/category.rs

```diff
@@ -156,6 +157,26 @@ impl Category {
         ).bind::<Text, _>(&self.category)
             .load(conn)
     }
+
+    /// Gathers the parent categories from the top-level Category to the direct parent of this Category.
+    /// Returns categories as a Vector in order of traversal, not including this Category.
+    /// The intention is to be able to have slugs or parent categories arrayed in order, to
+    /// offer the frontend, for examples, slugs to create links to each parent category in turn.
+    pub fn parent_categories(&self, conn: &PgConnection) -> QueryResult<Vec<Category>> {
+        use diesel::sql_types::Text;
+
+        sql_query(
+            "SELECT c.id, c.category, c.slug, c.description, \
```

Comment:
Can we move this whole thing into a `.sql` file and use `include_str!`, add some more newlines and indentation? The whole thing is very hard to parse. It's not obvious where the call to `COALESCE` ends for example

Reply:
:+1: Done.